src/rmodels.c

```diff
@@ -5614,6 +5623,9 @@ static Model LoadM3D(const char *fileName)
         // We always need a default material, so we add +1
         model.materialCount++;
 
+        // Sort faces by material.
+        qsort(m3d->face, m3d->numface, sizeof(m3df_t), m3d_compare_faces);
```

Comment:
I'd prefer to avoid using `qsort()` function, could the logic be implemented directly here?

Reply:
Sure, but I think it is already the most simple way

Reply:
It could be but it adds a function-dependency that I prefer to avoid.

Reply:
Ok, I've changed the sorting method to insertion

Reply:
@raysan5 I don't understand the objection to a `stlib.h` dependency.

Is it possible for an user to create a properly-sorted materials array and then deliver that to raylib in a straightforward manner, if that is what it will take?

Reply:
@orcmid There was a point the raylib development that I just tried to [minimize libc dependency](https://github.com/raysan5/raylib/wiki/raylib-libc-dependency) for raylib, it was a personal decision. I agree it probably has not much sense.